e2e_testing/torchscript/basic.py

```diff
@@ -918,10 +917,89 @@ def __init__(self):
         None,
         ([], torch.int64, True),
     ])
-
     def forward(self, input):
         return torch.numel(input)
 
 @register_test_case(module_factory=lambda: NumelZeroRankModule())
 def NumelZeroRankModule_basic(module, tu: TestUtils):
     module.forward(torch.randint(10,[]))
+
+
+class ZerosInt2DModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+    ])
+    def forward(self):
+        return torch.zeros(3, 4, dtype=torch.int64)
+
+@register_test_case(module_factory=lambda: ZerosInt2DModule())
+def ZerosInt2DModule_basic(module, tu: TestUtils):
+    module.forward()
+
+
+class ZerosInt3DModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+    ])
+    def forward(self):
+        return torch.zeros(3, 4, 5, dtype=torch.int64)
+
+@register_test_case(module_factory=lambda: ZerosInt3DModule())
+def ZerosInt3DModule_basic(module, tu: TestUtils):
+    module.forward()
+
+
+class ZerosFloat2DModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+    ])
+    def forward(self):
+        return torch.zeros(3, 4, dtype=torch.float32)
+
+@register_test_case(module_factory=lambda: ZerosFloat2DModule())
+def ZerosFloat2DModule_basic(module, tu: TestUtils):
+    module.forward()
+
+
+class ZerosFloat3DModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+    ])
+    def forward(self):
+        return torch.zeros(3, 4, 5, dtype=torch.float32)
+
+@register_test_case(module_factory=lambda: ZerosFloat3DModule())
+def ZerosFloat3DModule_basic(module, tu: TestUtils):
+    module.forward()
+
+
+class ZerosModuleFalsePinMemory(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+    ])
+    def forward(self):
+        return torch.zeros(3, 4, dtype=torch.float32, pin_memory=False)
+
+@register_test_case(module_factory=lambda: ZerosModuleFalsePinMemory())
+def ZerosModuleFalsePinMemory_basic(module, tu: TestUtils):
+    module.forward()
```

Comment:
This test case and the similar one for AtenOnesOp are failing. When I undo my changes, re-build, and test the case for AtenOnesOp, it passes. The original implementation https://github.com/llvm/torch-mlir/blob/main/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L3287-L3294 and my implementation https://github.com/llvm/torch-mlir/pull/464/commits/9f9970cccbd407d4867e4facebcc5c27a86ba44b#diff-fa644837a9cf28c5b4b9d88babd95dc19fc962b2036e083e4b79a6c9672982edR3294-R3301 both are identical. I don't know why the code is behaving differently. I tried hard to debug the error but could not resolve it. One thing I found that in my implementation the following condition https://github.com/llvm/torch-mlir/pull/464/commits/9f9970cccbd407d4867e4facebcc5c27a86ba44b#diff-fa644837a9cf28c5b4b9d88babd95dc19fc962b2036e083e4b79a6c9672982edR3295-R3299 evaluates to true, while for the original, it turns out to be false.

Reply:
@cathyzhyi @ramiro050